src/System.Private.CoreLib/shared/System/BitOps.cs

```diff
@@ -355,5 +355,210 @@ int SoftwareFallback(ulong v)
             }
 #endif
         }
+
+        /// <summary>
+        /// Reads whether the specified bit in a mask is set.
+        /// Similar in behavior to the x86 instruction BT.
+        /// </summary>
+        /// <param name="value">The value.</param>
+        /// <param name="bitOffset">The ordinal position of the bit to read.
+        /// Any value outside the range [0..7] is treated as congruent mod 8.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool ExtractBit(byte value, int bitOffset)
+            // For bit-length N, it is conventional to treat N as congruent modulo-N under the shift operation.
+            // So for uint, 1 << 33 == 1 << 1, and likewise 1 << -46 == 1 << +18.
+            // Note -46 % 32 == -14. But -46 & 31 (0011_1111) == +18. 
+            // So we use & not %.
+            => ExtractBit((uint)value, bitOffset & 7);
+
+        /// <summary>
+        /// Reads whether the specified bit in a mask is set.
+        /// Similar in behavior to the x86 instruction BT.
+        /// </summary>
+        /// <param name="value">The value.</param>
+        /// <param name="bitOffset">The ordinal position of the bit to read.
+        /// Any value outside the range [0..31] is treated as congruent mod 32.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool ExtractBit(uint value, int bitOffset)
+            => (value & (1u << bitOffset)) != 0;
+
+        /// <summary>
+        /// Reads whether the specified bit in a mask is set.
+        /// Similar in behavior to the x86 instruction BT.
+        /// </summary>
+        /// <param name="value">The value.</param>
+        /// <param name="bitOffset">The ordinal position of the bit to read.
+        /// Any value outside the range [0..7] is treated as congruent mod 8.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static bool ExtractBit(int value, int bitOffset)
```

Comment:
Per summary, note that all methods exposed with `byte`, `uint` and `int` overloads **only**.
Not sure if `long`, `ulong` should be provided too?